compiler/plugins/target/ROCM/builtins/ukernel/common.h

```diff
@@ -122,4 +122,37 @@ static inline uint64_t __ballot(int predicate) {
       predicate, 0, 33 /*ICMP_NE from llvm/include/llvm/IR/InstrTypes.h*/);
 }
 
+//===----------------------------------------------------------------------===//
+// Local definition for bf16 (Brain Floating Point 16-bit)
+//===----------------------------------------------------------------------===//
```

Comment:
Why is that needed? Since this code is only ever compiled with in-tree Clang/LLVM to AMDGPU target, which always supports the built-in type `__bf16`, you should be able to just use that.  This seems to confirm that it results in LLVM generating that conversion code for you:
https://godbolt.org/z/GreMEKhE6

The LLVM documention seems out of date in not mentioning support on AMDGPU,
https://clang.llvm.org/docs/LanguageExtensions.html#half-precision-floating-point